espnet2/enh/espnet_enh_s2t_model.py

```diff
@@ -179,6 +179,10 @@ def forward(
         # This is to handle flexible number of speakers.
         # Used only in "enh + diar" task for now.
         num_spk = text.shape[2] if text.dim() == 3 else self.enh_model.num_spk
+        if text.dim() == 3:
```

Comment:
Probably you may add the corresponding description about Enh + Diar in the function description information, especially about the text (meaning and shape, etc)